lib/vacations.ex

```diff
@@ -0,0 +1,143 @@
+defmodule GlassFactoryApi.Vacations do
+  @moduledoc """
+  Provides the methods to get vacations of all or a specific user.
+  """
+
+  alias GlassFactoryApi.ApiClient
+  alias GlassFactoryApi.Vacations.Vacation
+
+  @doc """
+  Return a tuple with `:ok` atom and an list of vacations.
+  or an tuple with `:error` and a string with the error description.
+
+  It's possible apply some filters, like user_id, start and end to limit dates, and vacation_type
+  using the query_string param.
+
+  ## Examples
+
+  iex> GlassFactoryApi.Clients.get_vacations()
+  {:ok,
+    [
+      %Vacation{
+        id: 119,
+        upcoming: false,
+        daily_hours: 8,
+        start_date: "2017-07-24",
+        end_date: "2017-07-30",
+        days: 7,
+        vacation_type: "paid",
+        creator_id: 722,
+        user_id: 722,
+      },
+      %Vacation{
+        id: 132,
+        upcoming: false,
+        daily_hours: 8,
+        start_date: "2019-07-20",
+        end_date: "2019-07-30",
+        days: 10,
+        vacation_type: "paid",
+        creator_id: 734,
+        user_id: 734
+      }
+    ]
+  }
+
+  iex> GlassFactoryApi.Clients.get_vacations([user_id: 734, start: 2019-07-01, end: 2019-07-31])
+  {:ok,
+    [
+      %Vacation{
+        id: 132,
+        upcoming: false,
+        daily_hours: 8,
+        start_date: "2019-07-20",
+        end_date: "2019-07-30",
+        days: 10,
+        vacation_type: "paid",
+        creator_id: 734,
+        user_id: 734
+      }
+    ]
+  }
+
+  iex> GlassFactoryApi.Clients.get_vacations("1")
+  {:ok, []}
+  """
+
+  @spec get_vacations([], map()) :: {atom(), [Vacation.t() | String.t()]}
+  def get_vacations(query_string \\ [], config \\ %{}) do
+    with {:ok, %{status: 200, body: body}} <- ApiClient.get("vacations", filter_query_string(query_string), config) do
```

Comment:
Same here, I'd favor `params` over `query_string`.

Reply:
Actually, in `ApiClient` it makes sense to call it `params`, at this level here we should probably call it `filters` or `opts`.